src/mon/OSDMonitor.cc

```diff
@@ -922,10 +922,12 @@ OSDMonitor::update_pending_pgs(const OSDMap::Incremental& inc)
     pending_creatings.pgs.erase(pg);
   }
   pending_created_pgs.clear();
-  // PAXOS_PGMAP is less than PAXOS_OSDMAP, so PGMonitor::update_from_paxos()
-  // should have prepared the latest pgmap if any
-  mon->pgservice->maybe_add_creating_pgs(creating_pgs.last_scan_epoch,
-					 &pending_creatings);
+  if (!osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
```

Comment:
If I read this correctly when updating, flagging this on LUMINOUS is exactly backwards. @tchaikov? I think we're trying to handle the overlap where the pgmap might have gotten an update which we didn't pick up in the separate creating_pgs set.

This also illustrates the main missing thing we need to deal with, which is the switch on upgrade from using the local pgmap to going via the ceph-mgr.
Also remaining: persistence of whatever state we need (from the mgr to the mon), and a sane way of dealing with stale data.